internal/cmd/commands/database/init.go

```diff
@@ -287,54 +298,45 @@ func (c *InitCommand) Run(args []string) (retCode int) {
 			return 1
 		}
 		if st.InitializationStarted {
-			// TODO: Separate from the "dirty" bit maintained by the schema
-			//  manager maintain a bit which indicates that this full command
-			//  was completed successfully (with all default resources being created).
-			//  Use that bit to determine if a previous init was completed
-			//  successfully or not.
 			c.UI.Error(base.WrapAtLength("Database has already been " +
 				"initialized. If the initialization did not complete successfully " +
 				"please revert the database to its fresh state."))
 			return 1
 		}
 	}
 
-	// This is an advisory locks on the DB which is released when the db session ends.
-	if err := man.ExclusiveLock(c.Context); err != nil {
-		c.UI.Error(fmt.Errorf("Error capturing an exclusive lock: %w", err).Error())
-		return 1
-	}
-	defer func() {
-		if err := man.ExclusiveUnlock(c.Context); err != nil {
-			c.UI.Error(fmt.Errorf("Unable to release exclusive lock to the database: %w", err).Error())
-		}
-	}()
-
-	migrationUrl, err := config.ParseAddress(migrationUrlToParse)
-	if err != nil && err != config.ErrNotAUrl {
-		c.UI.Error(fmt.Errorf("Error parsing migration url: %w", err).Error())
+	// The initialization is dirty as long as it doesn't complete successfully.
+	if err := man.SetDirty(c.Context); err != nil {
+		c.UI.Error(base.WrapAtLength("Database could not be marked as dirty in " +
+			"preperation for it's initialization."))
 		return 1
 	}
 
 	// Core migrations using the migration URL
 	{
 		migrationUrl = strings.TrimSpace(migrationUrl)
-		ran, err := schema.InitStore(c.Context, dialect, migrationUrl)
+		ran, err := schema.MigrateStore(c.Context, dialect, migrationUrl, schema.WithSkipSetDirty(true), schema.WithSkipUnsetDirty(true))
```

Comment:
Can you add a comment here and explain why you're skipping setting dirty and setting undirty? It's not clear to me especially given that you do call unsetDirty later, so if you're skipping setDirty...